### PR TITLE
Fixup 4.0 change log

### PR DESCRIPTION
**Pull request by dalg24**
Need to cherry-pick into 4.0.01 patch release

### CODE REVIEW

**Comment by crtrott** [comment]
Shouldn't most of these be there as "now removed"?

**Comment by dalg24** [comment]
> Shouldn't most of these be there as "now removed"?

If we listed the "now removed" then we'd need to expand the list quite a bit.  Also for most of these we are actually being gentle and buying users a bit more time.

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,7 @@
 - Make ExecutionSpace::concurrency() a non-static member function [\#5655](https://github.com/kokkos/kokkos/pull/5655) and related PRs
 
 ### Deprecations
+- Remove code guarded by `KOKKOS_ENABLE_DEPRECATED_CODE_3`
```
Comment (masterleinad) [inline CHANGELOG.md:77]: I would rather list this under "Incompatibilities".

Comment (dalg24) [reply CHANGELOG.md]: done

**Review by masterleinad — APPROVED** [review]

**Pull request merged**